modules/order/src/Plugin/Field/FieldWidget/AdjustmentDefaultWidget.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\commerce_order\Plugin\Field\FieldWidget;
+
+use Drupal\commerce_price\Plugin\Field\FieldWidget\PriceDefaultWidget;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Plugin implementation of the 'commerce_order_adjustment_default' widget.
+ *
+ * @todo Needs a functional test.
+ *
+ * @FieldWidget(
+ *   id = "commerce_order_adjustment_default",
+ *   label = @Translation("Adjustment"),
+ *   field_types = {
+ *     "commerce_order_adjustment"
+ *   }
+ * )
+ */
+class AdjustmentDefaultWidget extends PriceDefaultWidget {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    // We build the definition container and form, since it borrows items
+    // from the price widget. Then we add the selector, since we're passing
+    // the $element to the parent ::formElement.
+    $states_selector_name = $this->fieldDefinition->getName() . "[$delta][type]";
+    $element['definition'] = [
+      '#type' => 'container',
+      '#weight' => 2,
+      '#states' => [
+        'visible' => [
+          'select[name="' . $states_selector_name . '"]' => ['value' => 'order_adjustment'],
+        ],
+      ],
+    ];
+    $element['definition'] += parent::formElement($items, $delta, $element, $form, $form_state);
+
+    // Make sure the price title is "Amount".
+    $element['definition']['amount']['#title'] = $this->t('Amount');
+
+    $element['definition']['label'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Label'),
+      '#weight' => -10,
+      '#default_value' => $items[$delta]->label,
+    ];
+
+    $element['type'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Type'),
+      // @todo These options are hardcoded for now.
+      '#options' => [
+        '_none' => $this->t('- Select -'),
+        // @todo Promotion
+        // We want a specific way to add a promotion to an order.
+        // Not sure how this would look. Do we have an AdjustmentType plugin
+        // Which provides the configuration form? Or is that insane.
+        // Or.. is a promotion a reference field all on its own and not
+        // shown here?
+        // 'discount' => $this->t('Apply promotion')
+        'order_adjustment' => $this->t('Apply order adjustment'),
+        // @todo: How do we expose line item level adjustments on order UI.
```

Comment:
There are two options here. 
One is to have $line_item->adjustments, storing the line item adjustments directly on the line item. That requires the adjustment UI to be directly on the OrderEditForm, where it gathers the adjustments from the line items and the order, allows them to be edited.
The second option is to have all adjustments on the order, and track to what they apply (I believe that's what Spree does). That would allow the UI to be in the widget.